view/ce_dashboard/flask/blueprints/overview.py

```diff
@@ -0,0 +1,180 @@
+# for 1 hour. How does the caching for one hour work? The function is
+# decorated with @functools.lru_cache(maxsize=1), which caches the result
+# of the function call. The cache is invalidated after one hour by
+# using the hour_counter parameter. The function is called with the current
+# hour as the parameter, and the result is cached for that hour. When the
+# hour changes, the cache is invalidated and a new result is generated.
+@functools.lru_cache(maxsize=1)
+def _get_projects_tree(hour_counter: int) -> ET.Element:
+    hour_counter += 0  # just to avoid warning about unused variable
+    result_bytes, _ = _get_topology_projects_xml()
+    # Decode the bytes to a string using surrogateescape error handling
+    result_string = result_bytes.decode(
+        encoding="utf-8", errors="surrogateescape"
+    )
+    # Parse the XML response and return the root element
+    return ET.fromstring(result_string)
+
+# Cache the hash for one hours using the same hour_counter trick as above.
+# Cache is maxsize 2, so that we can cache both for field = "FieldOfScience"
+# and field = "Organization". 
+@functools.lru_cache(maxsize=2)
+def _get_project_fos_hash(hour_counter: int, field: str):
+    def safe_elem_text(elem: ET.Element) -> str:
+        try:
+            return elem.text.strip()
+        except AttributeError:  # we got a None
+            return ""
+
+    project_hash = {}
+
+    projects_tree = _get_projects_tree(hour_counter)
+    for project in projects_tree.findall("./Project"):
+        name = safe_elem_text(project.find("./Name"))
+        if not name:
+            continue  # malformed project?
+        field_of_science = safe_elem_text(project.find("./" + field))
+        project_hash[name] = field_of_science
+
+    return project_hash
+
+
+def organization_from_project(project_name: str):
+    hour_counter = int(time.time() / 3600)
+    return _get_project_fos_hash(hour_counter,"Organization").get(project_name, "Unknown").replace(',',' -')
+
+
+def field_of_science_from_project(project_name: str):
+    hour_counter = int(time.time() / 3600)
+    return _get_project_fos_hash(hour_counter,"FieldOfScience").get(project_name, "Unknown").replace(',',' -')
+
+
+#######################
+# Functions to query Ganglia Web and format response for HTCondorView
+#######################
+
+@cache_response_to_disk()
+def get_data_from_ganglia():
+    # Grab parameters host (CE name), r (range: day, week, or month)
+    host = request.args.get('host','chtc-spark-ce1.svc.opensciencegrid.org')
+    r = request.args.get('r','day')
+
+    # Grab data as a raw csv from ganglia, which was stashed there by the condor_gandliad
+    import pandas as pd
+    df=pd.read_csv('https://display.ospool.osg-htc.org/ganglia/graph.php?r=' + r + '&hreg[]=' + host + '&mreg[]=%5E' + 'Cpus' + '&mreg[]=%5E' + 'Memory' + '&mreg[]=%5E' + 'Disk' + '&mreg[]=%5E' + 'Bcus' + '&aggregate=1&csv=1',skipfooter=1,engine='python')
```

Comment:
Will this return an empty dataframe on a non-200 response? Or throw an exception?